### PR TITLE
feat: Added glue table indexes

### LINKED ISSUE #1366: Add a resolver that transforms array of objects with two fields to json
Some resources has field that has slice of objects with two fields. For example glue MLTransform has slice of column importances:
```
type ColumnImportance struct {

	// The name of a column.
	ColumnName *string

	// The column importance score for the column, as a decimal.
	Importance *float64

	noSmithyDocumentSerde
}
```
There is no need to create a separate table for structure above. The column can be represented as json with next structure
```
{
  <column_name>: <importannce>
}
```
To ease code generation `SliceToJsonPathResolver`  can be created  that takes "path", "key_path" and "value_path" and creates json from slice of objects.

### LINKED ISSUE #1368: AWS::Glue::Partition

### PR DESCRIPTION
**Pull request by amanenk**
#### Summary

closes #1368 
closes #1366 

---

Use the following steps to ensure your PR is ready to be reviewed

- [x] Read the [contribution guidelines](../blob/main/CONTRIBUTING.md) 🧑‍🎓
- [x] Run `go fmt` to format your code 🖊
- [x] Lint your changes via `golangci-lint run` 🚨 (install golangci-lint [here](https://golangci-lint.run/usage/install/#local-installation))
- [x] Update or add tests. Learn more about testing [here](https://docs.cloudquery.io/docs/developers/sdk/testing) 🧪
- [x] Update the docs by running `go run ./docs/docs.go` and committing the changes 📃
- [x] If adding a new resource, add [relevant Terraform files](../terraform) in a separate PR 📂
- [x] Ensure the status checks below are successful ✅

### CODE REVIEW

**Comment by disq** [comment]
Maybe add a unit test just for the resolver? The documentation (godoc) for the method needs more clarification or examples as well.

File: docs/tables/aws_glue_database_table_partitions.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+|database_table_cq_id|uuid|Unique CloudQuery ID of aws_glue_database_tables table (FK)|
+|account_id|text|The AWS Account ID of the resource.|
+|region|text|The AWS Region of the resource.|
+|catalog_id|text|The ID of the Data Catalog in which the partition resides|
+|creation_time|timestamp without time zone|The time at which the partition was created|
+|database_name|text|The name of the catalog database in which to create the partition|
+|last_access_time|timestamp without time zone|The last time at which the partition was accessed|
+|last_analyzed_time|timestamp without time zone|The last time at which column statistics were computed for this partition|
+|parameters|jsonb|These key-value pairs define partition parameters|
+|storage_descriptor_additional_locations|text[]|A list of locations that point to the path where a Delta table is located|
```
Comment (candiduslynx) [inline docs/tables/aws_glue_database_table_partitions.md:16]: `storage_descriptor_` prefix seems too much

File: docs/tables/aws_glue_database_table_partitions.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+|storage_descriptor_bucket_columns|text[]|A list of reducer grouping columns, clustering columns, and bucketing columns in the table|
+|storage_descriptor_compressed|boolean|True if the data in the table is compressed, or False if not|
+|storage_descriptor_input_format|text|The input format: SequenceFileInputFormat (binary), or TextInputFormat, or a custom format|
+|storage_descriptor_location|text|The physical location of the table|
+|storage_descriptor_number_of_buckets|bigint|Must be specified if the table contains any dimension columns|
+|storage_descriptor_output_format|text|The output format: SequenceFileOutputFormat (binary), or IgnoreKeyTextOutputFormat, or a custom format|
+|storage_descriptor_parameters|jsonb|The user-supplied properties in key-value form|
+|storage_descriptor_schema_reference_schema_id_registry_name|text|The name of the schema registry that contains the schema|
```
Comment (candiduslynx) [inline docs/tables/aws_glue_database_table_partitions.md:24]: `storage_descriptor_schema_reference_schema_id` -> json column?

File: docs/tables/aws_glue_database_table_partitions.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+|storage_descriptor_schema_reference_schema_id_schema_arn|text|The Amazon Resource Name (ARN) of the schema|
+|storage_descriptor_schema_reference_schema_id_schema_name|text|The name of the schema|
+|storage_descriptor_schema_reference_schema_version_id|text|The unique ID assigned to a version of the schema|
+|storage_descriptor_schema_reference_schema_version_number|bigint|The version number of the schema|
+|storage_descriptor_serde_info_name|text|Name of the SerDe|
```
Comment (candiduslynx) [inline docs/tables/aws_glue_database_table_partitions.md:29]: `storage_descriptor_serde_info` -> json column?

**Review by candiduslynx — CHANGES_REQUESTED** [review]

**Comment by bbernays** [comment]
I like the resolver you added in this PR... My question looking at these fields is whether this is going to fetch too much data... as all of the nested calls are going to happen sequentially.... And looking at the Index/Partition data it doesn't seem that compliance/ security focused...

**Comment by amanenk** [comment]
> I like the resolver you added in this PR... My question looking at these fields is whether this is going to fetch too much data... as all of the nested calls are going to happen sequentially.... And looking at the Index/Partition data it doesn't seem that compliance/ security focused...

Agree on partitions. I have removed them. I have left indexes because it is infrastructural data that does not depend on data that is "fed" to glue

**Review by candiduslynx — APPROVED** [review]

**Review by yevgenypats — APPROVED** [review]

**Review by bbernays — APPROVED** [review]

**Pull request merged**